lib/stdlib/src/epp.erl

```diff
@@ -1107,24 +1107,31 @@ scan_ifdef(Toks, IfDef, From, St) ->
     wait_req_skip(St, [ifdef]).
 
 scan_ifndef([{'(',_Alp},{atom,_Am,M},{')',_Arp},{dot,_Ad}], _IfnD, From, St) ->
-    case St#epp.macs of
-	#{M:=_Def} ->
+    case is_macro_defined(M, St) of
+        true ->
 	    skip_toks(From, St, [ifndef]);
-	_ ->
+        false ->
 	    scan_toks(From, St#epp{istk=[ifndef|St#epp.istk]})
     end;
 scan_ifndef([{'(',_Alp},{var,_Am,M},{')',_Arp},{dot,_Ad}], _IfnD, From, St) ->
-    case St#epp.macs of
-	#{M:=_Def} ->
+    case is_macro_defined(M, St) of
+        true ->
 	    skip_toks(From, St, [ifndef]);
-	_ ->
+        false ->
 	    scan_toks(From, St#epp{istk=[ifndef|St#epp.istk]})
     end;
 scan_ifndef(Toks, IfnDef, From, St) ->
     T = find_mismatch(['(',var_or_atom,')',dot], Toks, IfnDef),
     epp_reply(From, {error,{loc(T),epp,{bad,ifndef}}}),
     wait_req_skip(St, [ifndef]).
 
+is_macro_defined(Name, #epp{macs=Macs}) ->
+    case Macs of
+        #{Name := undefined} -> false;
```

Comment:
I hope this doesn't cause macros defined to `undefined` to be considered undefined. Does it?

```Erlang
-define(dummy, undefined).
-ifndef(dummy).
-error("Defined to undefined").
-endif.

Reply:
No, the value for your `dummy` macro would be represented as `[{atom,Anno,undefined}]`, which is different from `undefined` used to mark a pre-defined macro currently without a value.

Reply:
It's my understanding that defined macros are bound to `{none,[Expr]}` in that map, so there should be no issue.

Reply:
Good, so this can be resolved.